Refactor print_output

karlb opened the pull request:
The structure is easier to grasp now and the reduced duplication will make future changes easier.

karlb committed:
Remove redundant idx, idy, ... from print output

These matrices are now implied by the NaNs in the output.

karlb committed:
Force yhat to be float64 types

This makes dealing with the results more reliable. I'm not really sure
about the details, but this resolves some problems for me and seems
harmless enough.

karlb committed:
Don't modify stdout filehandle in print_output

karlb committed:
Use DataFrame.describe to summarize xvars,ymvars

No need to roll our own when this provides mostly the same output.

karlb committed:
Reduce redundancy when printing effects

karlb committed:
Further reduce redundancy when printing effects

karlb committed:
Remove yx_vars, yy_vars

The code is clearer without them.

Holger-Bartel reviewed (changes requested):
Seems fine. But could you please show me an example output, thanks.

karlb committed:
github actions: specify python version

Otherwise we can be silently upgraded to newer version where not all
dependencies are available, yet.

Holger-Bartel reviewed (approved):
Yes, indeed I just saw it before and it looks nice, thanks

karlb committed:
Fix pytorch version, increase min python version

The most recent pytorch version breaks compatibility. If we want to
upgrade, we should do so explicitly.

We don't test on python versions lower than 3.9, so we probably don't
have support for older versions any longer.
